tests/misc/test_serializable_inheritance.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+
+import ssz
+from ssz.sedes import (
+    uint8,
+)
+
+
+def create_inheritance_structure(structure):
+    class_definitions = tuple(definition.strip() for definition in structure.split())
+
+    class_dict = {}
+    field_names_dict = {}
+
+    for index, class_definition in enumerate(class_definitions):
+        # get the name of the class
+        class_name_start = 0
+        class_name_end = class_definition.index("(")
```

Comment:
Not necessary but it would be pretty trivial to use pyparsing to write this up as a simple grammar which would make the parsing a bit easier to understand.  Probably not needed in this PR but something I think I could do pretty quickly if we think readability of this code is important.

Reply:
Good idea! Didn't know about pyparsing, it seems like exactly the right tool for the job.